Major refactor

sansyrox opened the pull request:
## Major Refactor! Needs Help!

Todo

- [x] Convert Classes to Function
- [x] Add the original landing page to this version
- [ ] Move out Component from Video.js render method to individual file.
- [x] Convert CSS to Scss files
- [ ] Move the ConnectToSocketServer function in Video.js file to a separate utils folder.
- [ ] Add Recoil

Now, the server has been separated out to a different repo.

To run the server, git clone this repo: https://github.com/mexili/incogly-backend

and use `yarn dev` command in both the repos

you all can fork the branch `major_refactor` and make the prs to this branch only.

All the issues will be labelled as hard as refactors are harder in general.

sansyrox committed:
initiate major refactor

sansyrox committed:
Almost stable state

sansyrox committed:
Second stable version! Video call okayish, chat feature broken

sansyrox committed:
Major Refactor - Part 3

sansyrox committed:
Version Push 4

sansyrox committed:
Update

sansyrox committed:
Refactor complete

sansyrox commented:
@pranay-simejia , this is the refactor PR.

NishantJawla commented:
@sansyrox can I convert roompage.js from class-based component to a functional-based component

sansyrox commented:
@NishantJawla , if you take a look at this Pr, the room page.js file has been removed.

we have two class components here, we need to change them

Jassi10000-zz commented:
I work to work on #69 , pls assign me
@sansyrox sir I have mentioned the issue assignment here

nimit2801 commented:
@sansyrox is the Todo here our issue list for this PR?

sansyrox commented:
@nimit2801 , yes

sansyrox commented:
@Jassi10000 , that is not an active issue anymore. All the issues that we are focussing on atm are present in this PR's description.

Jassi10000-zz commented:
> @Jassi10000 , that is not an active issue anymore. All the issues that we are focussing on atm are present in this PR's description.

This thing I didn't knew as this was opened in the issues list , ok close it then , so that others also don't get confused

sansyrox commented:
> This thing I didn't knew as this was opened in the issues list , ok close it then , so that others also don't get confused

No worries 😄  Would still love your contributions here.

NishantJawla commented:
@sansyrox  I have opened a pull request regarding the CSS to scss file conversion please see it

NishantJawla committed:
Convert CSS to Scss files (#71)

Added node-sass@4.14.1 package to run SCSS. [The latest version of node-sass is not compatible]. changed files from CSS to SCSS and changed all the SCSS file address used

sansyrox commented:
> @sansyrox I have opened a pull request regarding the CSS to scss file conversion please see it

Merged!

nimit2801 commented:
hey, @sansyrox do we've to convert Class in Video.js and its methods to functional components?

sansyrox commented:
Yes @nimit2801

sansyrox committed:
Node Sass issue fixed

sansyrox committed:
Convert home component to functional

Jassi10000-zz reviewed (commented):
looks good

sansyrox committed:
Created home page || Need to include media library and improve responsiveness

sansyrox committed:
Fix buttons

sansyrox commented:
@Jassi10000 
> looks good

It certainly doesn't. Please don't write blind reviews.
Thanks!

Jassi10000-zz commented:
> @Jassi10000
> 
> > looks good
> 
> It certainly doesn't. Please don't write blind reviews.
> Thanks!

I just commented on the code that was changed to functional component , that was looking fine to me

raghavdhingra committed:
class to function component change

raghavdhingra committed:
setState callback fix

raghavdhingra committed:
small refactor

raghavdhingra committed:
Major refactor raghav (#77)

* class to function component change

* setState callback fix

* small refactor

raghavdhingra committed:
refactor video js

raghavdhingra committed:
Merge branch 'major_refactor' into major_refactor-raghav

raghavdhingra committed:
Merge pull request #78 from mexili/major_refactor-raghav

Major refactor raghav

sansyrox committed:
Moved all the inline styles to an external scss file

raghavdhingra committed:
code css refactored

raghavdhingra committed:
home page cleaning

raghavdhingra committed:
component cleaning

raghavdhingra committed:
component nomenclature

raghavdhingra committed:
repeated request issue fixed

raghavdhingra committed:
Update index.jsx

raghavdhingra committed:
code improvements

raghavdhingra committed:
Merge pull request #81 from mexili/reduce-socket-call

Reduce socket call

ZohebMOPO committed:
Improved UI style of the username input page (#88)

* Commiting to refactor-branch

* Distance added

* Same Size

* classnames changed

* BEM classnames

sansyrox committed:
Add design doc

raghavdhingra committed:
Create node.js.yml

raghavdhingra committed:
Update node.js.yml

raghavdhingra committed:
remove workflows

raghavdhingra committed:
fixed video stream on one end

raghavdhingra committed:
fixed multiple re-renders, made new refactored component

raghavdhingra committed:
refactored video component

sansyrox commented on src/components/Video/index_old.jsx:
```diff
@@ -0,0 +1,793 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
```
@raghavdhingra , do we need this index_old.jsx?

raghavdhingra replied:
If index.jsx file is tested thoroughly then we can delete the old jsx file @sansyrox

sansyrox commented:
Makes sense. Let us deploy the server and try to have the next convo on
incogly only. 🔥

On Sun, 20 Jun 2021 at 7:14 PM, Raghav Dhingra ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/components/Video/index_old.jsx
> <https://github.com/mexili/incogly/pull/67#discussion_r654938118>:
>
> > @@ -0,0 +1,793 @@
> +import React, { useCallback, useEffect, useRef, useState } from "react";
>
> If index.jsx file is tested thoroughly then we can delete the old jsx file
> @sansyrox <https://github.com/sansyrox>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mexili/incogly/pull/67#discussion_r654938118>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHEOIBSZVNLVKFZHPYWDMOTTTXWC7ANCNFSM4XPHMOXA>
> .
>

raghavdhingra commented:
Definitely 💪

sansyrox commented:
closing pr.

Made it the default
